File: python/datastructure/otterop/datastructure/_linked_list_iterator.py

class LinkedListIterator:
    
    def __init__(self, linked_list):
        self._current = linked_list.first()

    def has_next(self):
        return self._current != None

    def next(self):
        ret = self._current
        self._current = self._current.next()
        return ret.value()

File: ts/datastructure/otterop/datastructure/LinkedListIterator.ts
import { OOPIterator } from '@otterop/lang/OOPIterator';
import { LinkedListNode } from './LinkedListNode';
import { LinkedList } from './LinkedList';

/** @internal */
export class LinkedListIterator<T> implements OOPIterator<T> {

    #current : LinkedListNode<T>;

    /** @internal */
    constructor(linkedList : LinkedList<T>) {
        this.#current = linkedList.first();
    }

    public hasNext() : boolean {
        return this.#current != null;
    }

    public next() : T {
        let ret : LinkedListNode<T> = this.#current;
        this.#current = this.#current.next();
        return ret.value();
    }
}

